eipw-lint/src/lints/markdown/prevent_url.rs

```diff
@@ -0,0 +1,93 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+ use annotate_snippets::snippet::{Annotation, AnnotationType, Slice, Snippet};
+ use comrak::nodes::Ast;
+ use crate::lints::{Context, Error, Lint};
+ use crate::tree::{self, Next, TraverseExt};
+ use regex::Regex;
+ use serde::{Deserialize, Serialize};
+ use std::fmt::{Debug, Display};
+ 
+ #[derive(Debug, Serialize, Deserialize, Clone)]
+ pub struct PreventUrlsNoBackticks<S>(pub S);
```

Comment:
This should probably look something like:

```suggestion
 #[derive(Debug, Serialize, Deserialize, Clone)]
 pub struct PreventUrlsNoBackticks<S> {
     allowed_domains: Vec<S>,
 }
```

Then the default configuration in `lib.rs` would look like:

```rust
        (
            "markdown-prevent-url",
            MarkdownPreventUrl {
                allowed_domains: vec![
                    r"(.+\.)?example.com",
                    r"(.+\.)?invalid",
                    // And so on...
                ]
            }
        ), 
```